### PR TITLE
1203/instant apps

### PR DESCRIPTION
**Pull request by commjoen**
Thank you for submitting a Pull Request to the Mobile Security Testing Guide. Please make sure that:

- [x] Your contribution is written in the 2nd person (e.g. you)
- [x] Your contribution is written in an active present form for as much as possible.
- [x] You have made sure that the reference section is up to date (e.g. please add sources you have used, make sure that the references to MITRE/MASVS/etc. are up to date)
- [x] Your contribution has proper formatted markdown and/or code
- [x] Any references to website have been formatted as [TEXT](URL “NAME”)
- [x] You verified/tested the effectiveness of your contribution (e.g.: is the code really an effective remediation? Please verify it works!)

If your PR is related to an issue. Please end your PR test with the following line:
This PR covers issue #1203 & #1004.

### CODE REVIEW

**Comment by OMTGreleaser** [comment]
Results for commit cde6f39f809343f9778f1079603d0f0d69cf963c: Broken link result: [!] Error(s) found by the Links validator: 4 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:        6 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

**Comment by OMTGreleaser** [comment]
Results for commit ad4196d4bcc6e545c2fc7fbc8abbcd83e6b789cf: Broken link result: [!] Error(s) found by the Links validator: 5 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:        4 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

**Comment by OMTGreleaser** [comment]
Results for commit 55c2abd1a5572865e9da10e76cbfec83b6cd0f16: Broken link result: [!] Error(s) found by the Links validator: 5 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:        4 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

File: Document/0x05a-Platform-Overview.md [hunk 1]
```diff
@@ -141,6 +141,29 @@ In Android, the lifetime of an app process is controlled by the operating system
 - A cached process is a process that's not currently needed, so the system is free to kill it when memory is needed.
 Apps must implement callback methods that react to a number of events; for example, the `onCreate` handler is called when the app process is first created. Other callback methods include `onLowMemory`, `onTrimMemory` and `onConfigurationChanged`.
 
+##### App Bundles
+
+Android applications can be shipped in two forms: the Android Package Kit (APK) file or an [Android App Bundle](https://developer.android.com/guide/app-bundle) (.aab). Android App Bundles provide all the resources necessary for an app, but defer the generation of the APK and its signing to Google Play. App Bundles are signed binaries which contain the code of the app in several modules. The base module contains the core of the application. The base module can be extended with various modules which contain new enrichments/functionalities for the app as further explained at their [Documentation](https://developer.android.com/guide/app-bundle "Documentation on App Bundle").
```
Comment (TheDauntless) [inline Document/0x05a-Platform-Overview.md]: at their -> on the [developer documentation for app bundle]

File: Document/0x05a-Platform-Overview.md [hunk 2]
```diff
@@ -141,6 +141,29 @@ In Android, the lifetime of an app process is controlled by the operating system
...
 
+##### App Bundles
+
+Android applications can be shipped in two forms: the Android Package Kit (APK) file or an [Android App Bundle](https://developer.android.com/guide/app-bundle) (.aab). Android App Bundles provide all the resources necessary for an app, but defer the generation of the APK and its signing to Google Play. App Bundles are signed binaries which contain the code of the app in several modules. The base module contains the core of the application. The base module can be extended with various modules which contain new enrichments/functionalities for the app as further explained at their [Documentation](https://developer.android.com/guide/app-bundle "Documentation on App Bundle").
+If you have an Android App Bundle, you can best use the [Bundletool](https://developer.android.com/studio/command-line/bundletool "Bundle Tool") from Google to build unsigned APKs in order to use the existing tooling on the APK. You can create an APK from an AAB file by the following command:
```
Comment (TheDauntless) [inline Document/0x05a-Platform-Overview.md]: by the following command -> by running the following command

File: Document/0x05h-Testing-Platform-Interaction.md [hunk 3]
```diff
@@ -483,7 +483,7 @@ Consider this contrived example: `sms://compose/to=your.boss@company.com&message
 
 Once a URL scheme has been defined, multiple apps can register for any available scheme. For every application, each of these custom URL schemes must be enumerated and the actions they perform must be tested.
 
-URL schemes can be used for [deep linking](https://developer.android.com/training/app-links/ "Handling Android App Links"), a widespread and convenient way to launch a native mobile app via a link, which isn't inherently risky. Alternatively, since Android 6 App links can be used.
+URL schemes can be used for [deep linking](https://developer.android.com/training/app-links/ "Handling Android App Links"), a widespread and convenient way to launch a native mobile app via a link, which isn't inherently risky. Alternatively, since Android 6 App links can be used. App lnks, in contrast to deep links, require the domain of which the link is served to have a [digital asset link](https://developers.google.com/digital-asset-links/v1/getting-started "Digital Asset Link") and will ask the app to verify the asset-link first by means of using `android:autoVerify="true"` in the intentfilter.
```
Comment (TheDauntless) [inline Document/0x05h-Testing-Platform-Interaction.md]: Android 6 -> Android 6 (API level 23)

File: Document/0x05h-Testing-Platform-Interaction.md [hunk 4]
```diff
@@ -556,6 +556,61 @@ if (Intent.ACTION_VIEW.equals(intent.getAction())) {
 
 Defining and using your own URL scheme can be risky in this situation if data is sent to the scheme from an external party and processed in the app. Therefore keep in mind that data should be validated as described in "Testing custom URL schemes."
 
+### Testing for insecure Configuration of Instant Apps (MSTG-*)
```
Comment (TheDauntless) [inline Document/0x05h-Testing-Platform-Interaction.md]: I think the MSTG-* is messing up the markup

File: Document/0x05h-Testing-Platform-Interaction.md [hunk 5]
```diff
@@ -556,6 +556,61 @@ if (Intent.ACTION_VIEW.equals(intent.getAction())) {
...
+
+#### Overview
+
+With [Google Play Instant](https://developer.android.com/topic/google-play-instant/overview) you can now create Instant apps. An instant apps can be instantly launched from a browser or the "try now" button from the app store from Android API level 23 onward. They do not require any form of installation. There are a few challenges with instant apps:
```
Comment (TheDauntless) [inline Document/0x05h-Testing-Platform-Interaction.md]: An instant apps -> An instant app
Android API level 23 onward -> Android 6 (API level 23) onward

File: Document/0x05h-Testing-Platform-Interaction.md [hunk 6]
```diff
@@ -556,6 +556,61 @@ if (Intent.ACTION_VIEW.equals(intent.getAction())) {
...
+
+- there is a limited amount of size you can have with an instant app (max 10 mb).
```
Comment (TheDauntless) [inline Document/0x05h-Testing-Platform-Interaction.md]: Capitalize

File: Document/0x05h-Testing-Platform-Interaction.md [hunk 7]
```diff
@@ -556,6 +556,61 @@ if (Intent.ACTION_VIEW.equals(intent.getAction())) {
...
+- there is a reduced set of permissions it is allowed to use, which is documented at the [Android Instant app documentation](https://developer.android.com/topic/google-play-instant/getting-started/instant-enabled-app-bundle?tenant=irina#request-supported-permissions).
```
Comment (TheDauntless) [inline Document/0x05h-Testing-Platform-Interaction.md]: Ditto

Comment (TheDauntless) [reply Document/0x05h-Testing-Platform-Interaction.md]: there is a reduced set of permissions it is allowed to use -> Only a reduced number of permissions can be used, which are documented at

Comment (TheDauntless) [reply Document/0x05h-Testing-Platform-Interaction.md]: + Add alt text

File: Document/0x05h-Testing-Platform-Interaction.md [hunk 8]
```diff
@@ -556,6 +556,61 @@ if (Intent.ACTION_VIEW.equals(intent.getAction())) {
...
+
+The combination of these can lead to insecure decisions, such as: stripping too much of the authorization/authentication/confidentiality logic from an app, which allows for information leakage.
+
+Note: Instant apps require an app-bundle. App-bundles are described in the [App Bundle](#App Bundle) section.
+
+#### Static Analysis
+
+Static analysis can be either done after reverse engineering a downloaded instant app, or by analyzing the app bundle. When you analyze the app bundle: check the Android Manifest to see whether `dist:module dist:instant="true"` is set for a given module (either the base or a specific module with `dist:module` set). Next, check for the various entrypoints, which entrypoints are set (by means of `<data android:path="</PATH/HERE>" />`).
```
Comment (TheDauntless) [inline Document/0x05h-Testing-Platform-Interaction.md]: When you analyze the app bundle: check -> When you analyze the app bundle, check

File: Document/0x05h-Testing-Platform-Interaction.md [hunk 9]
```diff
@@ -556,6 +556,61 @@ if (Intent.ACTION_VIEW.equals(intent.getAction())) {
...
+
+Now follow the entrypoints, like you would do for any Activity and check:
+
+- is there any data retrieved by the app which should require privacy protection of that data? If so, are all required controls in place?
```
Comment (TheDauntless) [inline Document/0x05h-Testing-Platform-Interaction.md]: Capitals

Comment (TheDauntless) [reply Document/0x05h-Testing-Platform-Interaction.md]: ("Use appropriate capitalization and end punctuation for list items that individually form complete sentences.")

File: Document/0x05h-Testing-Platform-Interaction.md [hunk 10]
```diff
@@ -556,6 +556,61 @@ if (Intent.ACTION_VIEW.equals(intent.getAction())) {
...
+
+The installation of instant app support is taken care off through the following command:
+
+```shell
+$ cd path/to/android/sdk/tools/bin && ./sdkmanager 'extras;google;instantapps'
+```
+
+Next, you have to add `path/to/android/sdk/extras/google/instantapps/ia` to your `$PATH`.
+
+After the preparation, you can test instant apps locally on a device running Android 8.1 or later. You can
```
Comment (TheDauntless) [inline Document/0x05h-Testing-Platform-Interaction.md]: Android 8.1 -> Android 8.1 (API Level 27)

Comment (TheDauntless) [reply Document/0x05h-Testing-Platform-Interaction.md]: You can -> The app can be tested in different ways:

File: Document/0x05h-Testing-Platform-Interaction.md [hunk 11]
```diff
@@ -556,6 +556,61 @@ if (Intent.ACTION_VIEW.equals(intent.getAction())) {
...
+  ```
+
+- Test the app using the Play Console:
+  1. Upload your app bundle to the Google Play Console
+  2. Prepare the uploaded bundle for a release to the internal test track.
+  3. Sign into an internal tester account on a device, then launch your instant experience from either an external prepared link or via the `try now` button in the App store from the testers account.
+
+Now that you can test the app, check whether:
+
+- there are any data which require privacy controls and whether these controls are in place.
```
Comment (TheDauntless) [inline Document/0x05h-Testing-Platform-Interaction.md]: Capitals

**Comment by OMTGreleaser** [comment]
Results for commit d3bf643a6b97dbdd9fdb31430c1ddd1bfc4ccea2: Broken link result: [!] Error(s) found by the Links validator: 3 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:        5 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

**Comment by OMTGreleaser** [comment]
Results for commit f325626c0d6f433bf2fba8e4c514bbffdbbfd3c3: Broken link result: [!] Error(s) found by the Links validator: 3 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:        5 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

**Comment by OMTGreleaser** [comment]
Results for commit 13f829be7c46d2daddc2d52eea6d94cdc7747f3a: Broken link result: [!] Error(s) found by the Links validator: 3 pages have dead links! Verbose output in /link-check-result.out .
 Please run the Apply_Link_Check.sh in the tools folder to see wherhe the Link issues are. 
 Markdown result errorlength:        0 ../linter-result.out .
 Please run the Apply_Linter_check.sh in the tools folder to see where the Lint issues are.

**Pull request merged**